lib/chains.py

```diff
@@ -82,15 +82,16 @@ def generate_chains():
 
 def generate_chain_for_model(file_name, chains):
     model_name = file_name.split('.gguf')[0]
+    n_ctx = get_model_config(file_name)["loader_config"]["n_ctx"]
 
     chain = [None]
     llm_chain = lambda:  chain[-1] if chain[-1] is not None else chain.append(generate_llm_chain(file_name)) or chain[-1]
 
-    chains[model_name + ":core:text2text:summary"] = lambda: SummarizeChain(llm_chain=llm_chain())
+    chains[model_name + ":core:text2text:summary"] = lambda: SummarizeChain(llm_chain=llm_chain(), chunk_size=n_ctx)
     chains[model_name + ":core:text2text:headline"] = lambda: HeadlineChain(llm_chain=llm_chain())
     chains[model_name + ":core:text2text:topics"] = lambda: TopicsChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:simplification"] = lambda: SimplifyChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:formalization"] = lambda: FormalizeChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:reformulation"] = lambda: ReformulateChain(llm_chain=llm_chain())
+    # chains[model_name + ":core:text2text:simplification"] = lambda: SimplifyChain(llm_chain=llm_chain(), chunk_size=n_ctx)
+    # chains[model_name + ":core:text2text:formalization"] = lambda: FormalizeChain(llm_chain=llm_chain(), chunk_size=n_ctx)
+    # chains[model_name + ":core:text2text:reformulation"] = lambda: ReformulateChain(llm_chain=llm_chain(), chunk_size=n_ctx)
```

Comment:
Awesome, can we set this to 70% of the the context size so that there's still room for the response tokens?

Reply:
Done. Ideally, I'd like the percentage to be adjustable somehow, but I'm not sure if a user-adjustable setting would make sense either.

Reply:
We could also subtract the target summary size from the context size

Reply:
By "target summary size", I'm guessing you're referring to `max(len(inputs['input']) * 0.2, 1000)` in `summarize.py`?

Reply:
Yeah

Reply:
Done, but only on the summarize chain; I'm not sure if the other chains should behave similarly. Also, I made sure there is a minimum (positive) chunk size, which I temporarily set to 2048, but I'm not sure if that's an appropriate value either.